File: catalog/backends/php-laravel/patterns/SsrfExampleController.php
<?php

use Illuminate\Http\Request;
use Illuminate\Http\JsonResponse;

// Pattern: Code Security — SSRF via unvalidated URL parameter
// WARNING: intentionally vulnerable for IAST demo

function fetchUrl(Request $request): JsonResponse
{
    $url = $request->query('url', '');
    $content = file_get_contents($url);
    return response()->json(['body' => $content]);
}

File: catalog/backends/ruby-rails/patterns/ssrf_example_controller.rb
# Pattern: Code Security — SSRF via unvalidated URL parameter
# WARNING: intentionally vulnerable for IAST demo
# Adapt: use a domain-appropriate endpoint name

def fetch_url
  url = params[:url].to_s
  response = Net::HTTP.get(URI(url))
  render json: { body: response }
rescue StandardError => e
  render json: { error: e.message }, status: :internal_server_error
end
